src/algoesup/magics.py

```diff
@@ -159,6 +159,12 @@ def allowed(line: str) -> None:
     nargs="?",
     default=None,
 )
+@argument(
+    "-j",
+    "--jupyter",
+    help="Ignore irrelevant violations for Jupyter Notebook environments.",
+    action="store_true"
```

Comment:
Why a jupyter flag, since  `%ruff` can only be used in notebooks anyhow? Also, the configuration below actually selects rules that are not about Jupyter. What I'm saying is that the D100 W292 F401 and F821 rules should always be ignored, but other things can be added by the user in the command line, e.g. `%ruff --select A,B,C90,D,E,F,N,W,PL --ignore D203,D213,D415` would correspond to the current pyproject.toml.